pkg/labels/labels.go

```diff
@@ -417,6 +417,13 @@ func (l Labels) MergeLabels(from Labels) {
 	}
 }
 
+// Remove is similar to MergeLabels, but removes from the receiver object.
```

Comment:
:+1: yeah. Originally it did modify in-place, but I changed this because I think that this runs the risk of having multiple owners assume that they have control over a specific instance of the Labels object, whereas historically we have treated these as immutable structures. I think that your suggestion makes sense, will adopt it.